enterprise/server/cmd/ci_runner/main.go

```diff
@@ -1933,6 +1937,20 @@ func (ws *workspace) fetchTargetRef(ctx context.Context) error {
 	return ws.fetch(ctx, *targetRepoURL, []string{*targetBranch}, fetchDepth)
 }
 
+func (ws *workspace) updateSubmodules(ctx context.Context) error {
+	if _, err := os.Stat(".gitmodules"); err != nil {
+		if os.IsNotExist(err) {
+			// Repo doesn't have submodules.
+			return nil
+		}
+		return fmt.Errorf("stat .gitmodules: %w", err)
+	}
+	if _, err := git(ctx, ws.log, "submodule", "update", "--init", "--recursive"); err != nil {
```

Comment:
nit: This command doesn't seem to do anything if there are no submodules. Could skip the above check

Reply:
The command adds a bit of latency and I'm slightly worried about breaking people, so would prefer to keep the above check